x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java

```diff
@@ -241,6 +245,42 @@ public boolean isActionSafe(StepKey stepKey) {
         }
     }
 
+    /**
+     * Validate the name for an policy against some static rules. Intended to match
+     * {@link org.elasticsearch.cluster.metadata.MetaDataCreateIndexService#validateIndexOrAliasName(String, BiFunction)}
+     * @param policy the policy name to validate
+     * @throws IllegalArgumentException if the name is invalid
+     */
+    public static void validatePolicyName(String policy) {
+        if (!Strings.validFileName(policy)) {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not contain the following characters " +
+                Strings.INVALID_FILENAME_CHARS);
+        }
+        if (policy.contains("#")) {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not contain '#'");
+        }
+        if (policy.contains(":")) {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not contain ':'");
+        }
+        if (policy.charAt(0) == '_' || policy.charAt(0) == '-' || policy.charAt(0) == '+') {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not start with '_', '-', or '+'");
+        }
+        int byteCount = 0;
+        try {
+            byteCount = policy.getBytes("UTF-8").length;
+        } catch (UnsupportedEncodingException e) {
+            // UTF-8 should always be supported, but rethrow this if it is not for some reason
+            throw new ElasticsearchException("Unable to determine length of policy name", e);
+        }
+        if (byteCount > MAX_INDEX_NAME_BYTES) {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: name is too long, (" + byteCount + " > " +
+                MAX_INDEX_NAME_BYTES + ")");
+        }
+        if (policy.equals(".") || policy.equals("..")) {
+            throw new IllegalArgumentException("invalid policy name [" + policy + "]: must not be '.' or '..'");
+        }
```

Comment:
This feels like something we should have in a utility method shared between here and MetaDataCreateIndexService if we really intend to keep them in sync?

Reply:
@colings86 I've moved this logic out of here and MetaDataCreateIndexService into a shared utility class - does that look better?

Reply:
I have a general comment about the workflow here. Once we start refactoring an existing class that is core component of the system, that should become a separate PR, not a change done as part of a larger one, and not a change done in a feature branch that will be merged into master at some indeterminate point in the future. Separate changes should be separate PRs, and this will help increase the visibility of the change. A lot of folks will see ":Core/ILM" and immediately start ignoring.

Reply:
Makes sense, thanks @jasontedor.  I'll back the change out of this PR and make it in a separate PR if necessary based on the discussion in the issue.

Reply:
Thanks @gwbrown.